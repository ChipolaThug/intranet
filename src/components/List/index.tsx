import { memo } from "react"
import { Link } from "react-router-dom"

import { User } from "../../services/jsonPlaceholder"
import styles from "./styles.module.scss"

interface Props {
    items: User[]
}

const List = ({ items }: Props) => (
    <div className={styles.UserList}>
        <h4>User List</h4>
        <ul>
            {items.map(({ membreId, name }) => (
                <li key={membreId}>
                    <Link to={`/UserInfo/${membreId}`}>{name}</Link>
                </li>
            ))}
        </ul>
    </div>
)

export default memo(List)
