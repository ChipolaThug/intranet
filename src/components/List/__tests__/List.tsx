/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import List from "../index"

describe("<List />", () => {
    it("renders", () => {
        const tree = render(
            <MemoryRouter>
                <List
                    items={[
                        {
                            membreId: 1,
                            name: "PeL",
                            phone: "+886 0970...",
                            email: "forceoranj@gmail.com",
                            website: "https://www.parisestludique.fr",
                        },
                    ]}
                />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
