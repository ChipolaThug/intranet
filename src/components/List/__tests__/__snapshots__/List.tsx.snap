// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> renders 1`] = `
<div
  class="UserList"
>
  <h4>
    User List
  </h4>
  <ul>
    <li>
      <a
        href="/UserInfo/1"
      >
        PeL
      </a>
    </li>
  </ul>
</div>
`;
